zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/tasks/incident/IncidentUpdateRepositoryIT.java

```diff
@@ -798,13 +799,106 @@ void shouldNotGetFlowNodesWithOtherKeys() throws PersistenceException {
+      batchRequest.addWithId(
+          incidentTemplate.getFullQualifiedName(),
+          "8",
+          newIncident(8).setState(IncidentState.ACTIVE).setTreePath("PI_1/FNI_2/PI_3/FNI_4"));
+      batchRequest.addWithId(
+          incidentTemplate.getFullQualifiedName(),
+          "9",
+          newIncident(9).setState(IncidentState.ACTIVE).setTreePath("PI_5/FNI_6"));
+      batchRequest.executeWithRefresh();
+
+      // when
+      final var incidents = repository.getActiveIncidentsByTreePaths(List.of("PI_1"));
+
+      // then
+      assertThat(incidents)
+          .succeedsWithin(Duration.ofSeconds(10))
+          .asInstanceOf(InstanceOfAssertFactories.collection(ActiveIncident.class))
+          .containsExactlyInAnyOrder(
+              new ActiveIncident("8", "PI_1/FNI_2/PI_3/FNI_4"),
+              new ActiveIncident("7", "PI_1/FNI_2"));
+    }
+
+    @Test
+    void shouldNotReturnInactiveIncidents() throws PersistenceException {
+      // given
+      final var repository = createRepository();
+      final var batchRequest = clientAdapter.createBatchRequest();
+      batchRequest.addWithId(
+          incidentTemplate.getFullQualifiedName(),
+          "7",
+          newIncident(7).setState(IncidentState.ACTIVE).setTreePath("PI_1/FNI_2"));
+      batchRequest.addWithId(
+          incidentTemplate.getFullQualifiedName(),
+          "8",
+          newIncident(8).setState(IncidentState.PENDING).setTreePath("PI_1/FNI_3"));
+      batchRequest.addWithId(
+          incidentTemplate.getFullQualifiedName(),
+          "9",
+          newIncident(9).setState(IncidentState.RESOLVED).setTreePath("PI_1/FNI_4"));
+      batchRequest.executeWithRefresh();
+
+      // when
+      final var incidents = repository.getActiveIncidentsByTreePaths(List.of("PI_1"));
+
+      // then
+      assertThat(incidents)
+          .succeedsWithin(Duration.ofSeconds(10))
+          .asInstanceOf(InstanceOfAssertFactories.collection(ActiveIncident.class))
+          .containsExactlyInAnyOrder(new ActiveIncident("7", "PI_1/FNI_2"));
+    }
+
+    @Test
+    void shouldGetActiveIncidents() throws PersistenceException {
+      // given
+      final var repository = createRepository();
+      final var batchRequest = clientAdapter.createBatchRequest();
+      batchRequest.addWithId(
+          incidentTemplate.getFullQualifiedName(),
+          "7",
+          newIncident(7).setState(IncidentState.ACTIVE).setTreePath("PI_1/FNI_2"));
+      batchRequest.addWithId(
+          incidentTemplate.getFullQualifiedName(),
+          "8",
+          newIncident(8).setState(IncidentState.ACTIVE).setTreePath("PI_3/FNI_4"));
+      batchRequest.addWithId(
+          incidentTemplate.getFullQualifiedName(),
+          "9",
+          newIncident(9).setState(IncidentState.ACTIVE).setTreePath("PI_5/FNI_6"));
+      batchRequest.executeWithRefresh();
+
+      // when
+      final var incidents = repository.getActiveIncidentsByTreePaths(List.of("PI_1", "PI_3"));
```

Comment:
:question:  All the TreePaths used as arguments have only depth equal to one, so I would use something like this:
```List.of("PI_1", "PI_3/FNI_4", "PI_5/FNI_6/FN_4")```
The last TreePath also tests deeper TreePaths than the incident